app/parse.py

```
@@ -14,10 +22,108 @@ class Product:
     rating: int
     num_of_reviews: int
 
+def scrape_products_from_page(driver: webdriver.Chrome, url: str, product_selector: str, use_more_button: bool = False) -> List[Product]:
+    driver.get(url)
+    wait = WebDriverWait(driver, 10)
+
+    try:
+        cookies = driver.find_element(By.CSS_SELECTOR, ".accept-cookies-button")
+        if cookies:
+            cookies[0].click()
```

Cookie handling is incorrect and will raise an error. You call `driver.find_element(...)` (line 30) which returns a single WebElement, then attempt to index it with `[0]` on line 32. That indexing will raise a TypeError. Also you catch all exceptions with a broad `except Exception:` (line 33) which hides real problems and may mask other errors.

- Fix: use `find_elements(By.CSS_SELECTOR, ".accept-cookies-button")`, check the returned list length, and click the first element if present. If you use a wait, catch `TimeoutException` specifically. Avoid `except Exception:` so real errors surface.